Automattic/jetpack
# Story block: Improve rendering in email notifications

This PR attempts to fix 2 issues with email notifications of blog posts containing a story block on WPCOM:
- Fix scaling the image to 180x320
- Fix story preview link

<img width="485" alt="sh-merged" src="https://user-images.githubusercontent.com/230230/117172896-89194380-adcc-11eb-9138-f60b3d43bf55.png">

#### Changes proposed in this Pull Request:
This PR updates some CSS rules to makes sure the story blocks renders ok in preview mode, even without javascript. This fixes an issue where the first image in a story would not properly scale to the size of the story.
We also update the whole story rendering logic on the server to reuse the CSS classes but with a different DOM structure, more suited for consumption without javascript (the whole story is an anchor tag for instance).

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
TBD

* Apply D61026-code to your sandbox
* Find the blog id and post id of a story you want to test
* From your sandbox run `php ./bin/subscriptions/send.php BLOG_ID post POST_ID YOUR_EMAIL_ADDRESS`
* Check the email in different clients (gmail had the issue before this fix), the image should be scaled to the size of the preview (180x320), since https://github.com/Automattic/jetpack/pull/18609 the full size image is loaded
* Click on the image, your email client should open a new tab which loads the story in fullscreen
* Disable javascript and reload the page without the query args
* The story should look the same (in embed format), clicking on it should open the same page with the query arguments needed to view it fullscreen
* Go to the WPCOM reader, and find a story post, make sure the image is propely scaled (it's actually using a 180x320px image, so no "retina" support yet) and that it still has a link to the playable story
